week4_ts/src/components/Header.tsx

```diff
@@ -0,0 +1,144 @@
+            new Promise((resolve) => {
+                navigator.geolocation.getCurrentPosition((currentPosition) => {
+                    position.current = currentPosition.coords;
+                    const params = {
+                        y: position.current.latitude,
+                        x: position.current.longitude,
+                        radius: 1000,
+                        query: "베이커리",
+                    };
+                    storeSearchHttpHandler(params);
+                });
+            });
+        } else {
+            const params = {
+                x: position.current.longitude,
+                y: position.current.latitude,
+                radius: 1000,
+                query: "바",
+            };
+            storeSearchHttpHandler(params);
+        }
+    };
+
+    // 클릭 시 검색 input 비활성화
+    const handleInputDisabled = () => {
+        if (searchRef.current !== null) {
+            searchRef.current.disabled = !searchRef.current.disabled;
+            setIsLocation((prev) => !prev);
+        }
+    };
+    // input 변화 감지
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setInput(e.target.value);
+    };
+    // 검색 버튼
+    const handleSearchButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+        handleIsSearch(true);
+        e.preventDefault();
+        if (searchRef.current !== null) {
+            if (!searchRef.current.disabled) {
+                const params = {
+                    query: input + " " + "바",
+                };
+                storeSearchHttpHandler(params);
+            } else {
+                handleMyLocation();
+            }
+        }
+    };
+
+    return (
+        <Styled.Root>
+            <h1>우리 동네 BAR 🥂</h1>
+            <Styled.SearchWrapper>
+                <MyLocationButton isChoice={isLocation} onClick={handleInputDisabled}>
+                    현위치로 할래
+                </MyLocationButton>
+                <SearchLabel>찾고싶은 동네는</SearchLabel>
+                <SearchInput
+                    ref={searchRef}
+                    type="text"
+                    onChange={handleInputChange}
+                    value={input}
+                    placeholder="지역을 입력해주세요"
+                />
+                <SearchButton isChoice={handleIsSearch} type="submit" onClick={handleSearchButton}>
+                    검색
+                </SearchButton>
+            </Styled.SearchWrapper>
+        </Styled.Root>
+    );
+}
+
+export default Header;
+
+interface StHeaderProps {
+    isChoice: boolean;
+}
+
+const Styled = {
+    Root: styled.header`
+        ${flexColumnCenter}
+        & h1 {
+            margin-bottom: 2rem;
+        }
+    `,
+    SearchWrapper: styled.div`
+        display: flex;
+        gap: 2rem;
+    `,
+};
+
+const MyLocationButton = styled.button`
+    // 왜 색 안바뀌지
+    color: ${(props: StHeaderProps) =>
+        props.isChoice ? theme.colors.skyblue : theme.colors.lightgreen};
+`;
```

Comment:
왜 색 안 바뀌는지는 모르겠지만,,, 아래처럼도 SC props 타입 지정이 가능합니다!
제네릭으로 선언함으로써, 이후 css에서도 isChoice를 사용할 때마다 타입지정을 하지 않아도 됩니다

```suggestion
const MyLocationButton = styled.button<{ isChoice: boolean }>`
    color: ${({ isChoice }) =>
        isChoice ? theme.colors.skyblue : theme.colors.lightgreen};
`;
```

Reply:
```ts
const colors = {
    black: "#333333",
    white: "#ffffff",
    lemonchiffon: "#fffacd",
    gray: "#6B7280",
    skeleton: "#808080",
    skyblue: "87ceeb",
    lightgreen: "99f299",
};
```

# 붙여주세요,,,,,